contracts/staking/contracts/src/staking_pools/MixinStakingPool.sol

```diff
@@ -64,191 +44,83 @@ contract MixinStakingPool is
-        if (numMakersByPoolId[poolId] == maximumMakersInPool) {
-            LibRichErrors.rrevert(LibStakingRichErrors.MakerPoolAssignmentError(
-                LibStakingRichErrors.MakerPoolAssignmentErrorCodes.PoolIsFull,
-                makerAddress,
-                poolId
-            ));
-        }
-
-        // Add maker to pool
-        IStructs.MakerPoolJoinStatus memory poolJoinStatus = IStructs.MakerPoolJoinStatus({
-            poolId: poolId,
-            confirmed: true
-        });
-        poolJoinedByMakerAddress[makerAddress] = poolJoinStatus;
-        numMakersByPoolId[poolId] += 1;
-
-        // Maker has been added to the pool
-        emit MakerAddedToStakingPool(
+        // load pool and assert that we can decrease
+        uint32 currentOperatorShare = poolById[poolId].operatorShare;
+        _assertNewOperatorShare(
             poolId,
-            makerAddress
+            currentOperatorShare,
+            newOperatorShare
         );
-    }
-
-    /// @dev Removes a maker from a staking pool. Note that this is only callable by the pool operator or maker.
-    /// Note also that the maker does not have to *agree* to leave the pool; this action is
-    /// at the sole discretion of the pool operator.
-    /// @param poolId Unique id of pool.
-    /// @param makerAddress Address of maker.
-    function removeMakerFromStakingPool(
-        bytes32 poolId,
-        address makerAddress
-    )
-        external
-        onlyStakingPoolOperatorOrMaker(poolId, makerAddress)
-    {
-        bytes32 makerPoolId = getStakingPoolIdOfMaker(makerAddress);
-        if (makerPoolId != poolId) {
-            LibRichErrors.rrevert(LibStakingRichErrors.MakerPoolAssignmentError(
-                LibStakingRichErrors.MakerPoolAssignmentErrorCodes.MakerAddressNotRegistered,
-                makerAddress,
-                makerPoolId
-            ));
-        }
-
-        // remove the pool and confirmation from the maker status
-        IStructs.MakerPoolJoinStatus memory poolJoinStatus = IStructs.MakerPoolJoinStatus({
-            poolId: NIL_POOL_ID,
-            confirmed: false
-        });
-        poolJoinedByMakerAddress[makerAddress] = poolJoinStatus;
-        numMakersByPoolId[poolId] -= 1;
 
-        // Maker has been removed from the pool`
-        emit MakerRemovedFromStakingPool(
+        // decrease operator share
+        poolById[poolId].operatorShare = newOperatorShare;
+        emit OperatorShareDecreased(
             poolId,
-            makerAddress
+            currentOperatorShare,
+            newOperatorShare
         );
     }
 
-    /// @dev Returns the pool id of the input maker.
-    /// @param makerAddress Address of maker
-    /// @return Pool id, nil if maker is not yet assigned to a pool.
-    function getStakingPoolIdOfMaker(address makerAddress)
+    /// @dev Returns the unique id that will be assigned to the next pool that is created.
+    /// @return Pool id.
+    function getNextStakingPoolId()
```

Comment:
This function can be removed since `nextPoolId` is now public (reminder to remove from any interfaces as well).